core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -743,8 +743,11 @@ private[spark] class SparkSubmit extends Logging {
       }
     }
 
-    // In case of shells, spark.ui.showConsoleProgress can be true by default or by user.
-    if (isShell(args.primaryResource) && !sparkConf.contains(UI_SHOW_CONSOLE_PROGRESS)) {
+    // In case of shells, spark.ui.showConsoleProgress can be true by default or by user. Except,
+    // when Spark Connect is in local mode, because Spark Connect support its own progress
+    // reporting.
+    if (isShell(args.primaryResource) && !sparkConf.contains(UI_SHOW_CONSOLE_PROGRESS) &&
+      !sparkConf.contains("spark.local.connect")) {
```

Comment:
```suggestion
        !sparkConf.contains("spark.local.connect")) {
```

Reply:
So I'm not sure this is the proper indentation, my intellij with the scalafmt profile says it is correct like this 🤷 Yay for auto format